src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,125 @@
+            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name) &&
+                m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+
+            foreach (var untestedMember in untestedMembers)
+            {
+                context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                    displayText: untestedMember.Name.EscapeIdentifier(),
+                    insertionText: null,
+                    symbols: ImmutableArray.Create(untestedMember),
+                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    rules: s_rules));
+            }
+        }
+
+        private static bool IsFieldOrReadableProperty(ISymbol symbol)
+        {
+            if (symbol.IsKind(SymbolKind.Field))
+            {
+                return true;
+            }
+
+            if (symbol.IsKind(SymbolKind.Property) && !((IPropertySymbol)symbol).IsWriteOnly)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Create(enterKeyRule: EnterKeyRule.Never);
+
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+            => CompletionUtilities.IsTriggerCharacter(text, characterPosition, options) || text[characterPosition] == ' ';
+
+        private static SyntaxToken TryGetOpenBraceOrCommaInPropertyPattern(
+            Document document, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
+        {
+            var tree = semanticModel.SyntaxTree;
+            if (tree.IsInNonUserCode(position, cancellationToken))
+            {
+                return default;
+            }
+
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            if (!token.IsKind(SyntaxKind.CommaToken, SyntaxKind.OpenBraceToken))
+            {
+                return default;
+            }
+
+            return (token.Parent.IsKind(SyntaxKind.PropertySubpattern) == true) ? token : default;
+        }
+
+        // List the members that are already tested in this property sub-pattern
+        private static HashSet<string> GetTestedMembers(SyntaxTree tree, int position, CancellationToken cancellationToken)
+        {
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken)
+                .GetPreviousTokenIfTouchingWord(position);
+
+            // We should have gotten back a { or ,
+            if (token.IsKind(SyntaxKind.CommaToken, SyntaxKind.OpenBraceToken))
```

Comment:
also get rid of this check, because this was already done inside `TryGetOpenBraceOrCommaInPropertyPattern`. the token you get from that will always satisfy this

Reply:
if you want to make this assumption clear inside this method, you could name the token `openBraceOrCommaToken` or even add an assert